### ISSUE TITLE
Remove Open as Admin

### ISSUE THREAD

**Issue by tintou** [elementary/files#224]
This desktop action should go away to impeach users without knowledge to fuck their system

**Comment by SubhadeepJasu** [comment]
When people select this option, they know what they are doing. If it f**ks up their system then they are the ones who are responsible, Also elementary usually gives an warning in the form of authentication. I also feel like a little more elaborate warning as what this action can do might of help to absolute rookies.

**Comment by worldofpeace** [comment]
For example. You click 'New Window As Administrator' -> It opens and displays a MessageDialog warning you about the dangers.

**Comment by tintou** [comment]
Here normal user have no reason to do this, user with further needs can still use "pkexec pantheon-files" or even execute the commands mv/cp/rm from a command line with elevated rights

**Comment by worldofpeace** [comment]
It's a nice thing to have for people with more advanced knowledge to make their workflow easier. 
I think it's good to assume that people are competent, and capable but sometimes not knowledgeable.

They can always run ``rm -rf /`` in the terminal to see what it does :). 
No one is exempt from messing up their system.

**Comment by Tidrek** [comment]
Not having to use terminal for typical tasks is an indication of an elaborate desktop environment. What you propose is a step back. A scary authentication window with a warning is more than enough. If a user purposefully opened a root window, entered password an "accidentally" deleted few too important files, it's not a developer's fault.

I opened an issue - a proposal for improvement for Scratch to include a similar option like that. Someone implemented it and made a pull request but now it's held back because of this. Damn shame :(

**Comment by jeremypw** [comment]
Note this old branch: https://code.launchpad.net/~jeremywootten/pantheon-files/fix-1631982-run-as-admin-warning.

It has got left behind when we moved to GitHub but could be resurrected.

**Comment by Tidrek** [comment]
So, is there any progress in the discussion?

**Comment by worldofpeace** [comment]
@Tidrek I don't know. 

I would like elementary OS to more transparent on what design changes they're making in our OS :100: Don't leave us in the dark. We want to know what's going on as well.

**Comment by jeremypw** [comment]
No decision has been made on this issue and it is not under active discussion at the moment.

**Comment by liqsquid00** [comment]
Why not use the standard approach and leave the advanced options under the shift key?

Example: if the file is only clicked with the right button, the normal context menu show up with the options "Open", "Move to trash" etc., but if the file is clicked holding shift or if shift is pressed as the context menu is open, the advanced options "Open as administrator" and "Delete" substitute the unprivileged ones or simply are added as new options.

I see no reason as to completely remove power options. A simple, secure (for users without knowledge) and clean program doesn't need to be a crippled one.

One more suggestion I have is to standardize the authentication window requiring administrator rights, making it immediately recognizable system-wide as "ow, this is serious.", possible with a distinctive artwork showing in it; something that speaks for itself and is easy to be remembered like the "Aw, Snap!" folder picture in Google Chrome browser

**Comment by Tidrek** [comment]
Recently in KDE they removed this option from dolphin. Now it's even impossible to run it as root from console. They said it's because of security issues that a file manager should never be run as root. But it's not so bad, because in Dolphin when you press F4, a console pops in inside the dolphin window. When you move from folder to folder, it changes path automatically. Then you press F4 again and it disappears.

The key feature is that it's just one button press, not right click -> open terminal here, and it's not a separate window, that I have to manage separately, it's a part of the file manager.

But something tells me, it will be much harder to implement this in eOS than it was in KDE and the quickest route for now is to give people what they want.

**Comment by jlnr** [comment]
Running UI applications as root is [not really supported under Wayland](https://ubuntuforums.org/showthread.php?t=2374812) anyway.

I'm not sure how the proposed replacement (PolicyKit) works, but I guess it could work as on macOS -- if the user tries to make a filesystem change for which they lack permissions, Files could automatically open an auth prompt, and the actual operation then happens in a process with root permissions. Basically sudo for GUI applications.

**Comment by SubhadeepJasu** [comment]
You are right, Files as admin is basically the same as Files as sudo. Also for some reason the normal FIles ahows the root file system too but without write permissions to most of it. It would be a good idea to either hide the Root filesystem in the normal instance of Files or as @jlnr said, only prompt for password when any changes to such files are made to which normal access is denied.

**Comment by jeremypw** [comment]
Note that you can now use the gvfs admin backend as an alternative to opening Files as administrator.  Typing `admin://` into the pathbar opens a tab with admin permissions.
